src/offline/mod_offline_api.erl

```diff
@@ -2,33 +2,33 @@
 
 -export([delete_expired_messages/1, delete_old_messages/2]).
 
--spec delete_expired_messages(jid:lserver()) ->
-    {ok | domain_not_found | server_error, iolist()}.
+-type(api_result()) :: {ok, string()} | {domain_not_found | server_error, string()}.
+
+-spec delete_expired_messages(jid:lserver()) -> api_result().
 delete_expired_messages(Domain) ->
-    call_for_loaded_module(Domain, fun remove_expired_messages/2, {Domain}).
+    call_for_loaded_module(Domain, fun remove_expired_messages/2, [Domain]).
```

Comment:
If you are refactoring this one, changing the name would be nice, because it is not checking that the module is loaded - it only checks if the host type exists.